Clustering/hybridEcalSplitter/src/DDStripSplitter.cc

```diff
@@ -0,0 +1,728 @@
+      break;
+    default:
+      streamlog_out ( ERROR ) << "ERROR crazy stuff!!! abandoning event..." << endl;
+      return;
+    }
+
+    for (uint i=0; i<toSplit->size(); i++) { // loop over collections of this type (long/trans)
+      try {
+	LCCollection * col = evt->getCollection( toSplit->at(i).c_str() );
+	if (!col) continue;
+
+	const std::string layerCodingString(col->getParameters().getStringVal(LCIO::CellIDEncoding));
+
+	// is this a barrel or endcap collection?
+	TString hhname = toSplit->at(i);
+	bool barrel(true);
+	if (hhname.Contains("Barrel") || hhname.Contains("barrel")) {
+	  barrel = true;
+	} else if (hhname.Contains("Endcap") || hhname.Contains("endcap")) {
+	  barrel = false;
+	} else {
+	  streamlog_out ( ERROR ) << "WARNING: cannot tell if collection is for barrel or endcap..." << hhname << endl;
+	}
+
+	// make new collections for split and unsplit strips
+	IMPL::LCCollectionVec* splitStripHits = new IMPL::LCCollectionVec( LCIO::CALORIMETERHIT );
+	splitStripHits->setFlag(splitStripHits->getFlag()|( 1 << LCIO::RCHBIT_LONG)); // store position
+	splitStripHits->parameters().setValue(LCIO::CellIDEncoding, layerCodingString);
+
+	IMPL::LCCollectionVec* unSplitStripHits = new IMPL::LCCollectionVec( LCIO::CALORIMETERHIT );
+	unSplitStripHits->setSubset();
+	unSplitStripHits->parameters().setValue(LCIO::CellIDEncoding, layerCodingString);
+	
+	// get the cellid decoder for this collection
+	CellIDDecoder<CalorimeterHit> id( col ) ;
+	_decoder = &id;
+	
+	// loop over the collection's hits
+	int nelem = col->getNumberOfElements();
+	for (int j=0; j < nelem; ++j) {
+	  CalorimeterHit * hit = dynamic_cast<CalorimeterHit*>(col->getElementAt(j) );
+	  if (!hit) {
+	    streamlog_out ( ERROR ) << "ERROR  null hit in collection " << toSplit->at(i).c_str() << " " << j << endl;
+	    continue;
+	  }
+	  // split the hits
+	  std::vector <CalorimeterHit*> splitHits = getVirtualHits(evt, hit, orientation, barrel);
+
+	  // add (new) hits to collections
+	  if (splitHits.size()==0) { // not split, add original hit
+	    unSplitStripHits->addElement(hit);
+	  } else { // split was split, add the virtual hits
+	    for (uint hh=0; hh<splitHits.size(); hh++) {
+	      splitStripHits->addElement(splitHits[hh]);
+	    }
+	  }
+
+	} // loop over hits in collection
+
+	if (splitStripHits->getNumberOfElements()>0)
+	  outputcolls[splitStripHits] = toSplit->at(i) + "Split";
+	if (unSplitStripHits->getNumberOfElements()>0)
+	  outputcolls[unSplitStripHits] = toSplit->at(i) + "UnSplit";
+
+      } catch(DataNotAvailableException &e) {};
+    } // loop over collections
+  } // long/trans loop
+
+  std::map < IMPL::LCCollectionVec*, std::string >::iterator ii;
+
+  // add the new collections to the event
+  for (ii=outputcolls.begin(); ii!=outputcolls.end(); ii++) {
+    evt->addCollection(ii->first, ii->second);
+  }
+
+  if (_saveIntersections) {
+    evt->addCollection(intersectionHits,"stripIntersections");
```

Comment:
Can you add processor parameters for these 3 collections ?

Reply:
done